src/kdbindings/signal.h

```diff
@@ -239,12 +84,48 @@ class Signal
         // value can be used to disconnect the function again.
         Private::GenerationalIndex connect(std::function<void(Args...)> const &slot)
         {
-            return m_connections.insert({ slot });
+            Connection newConnection;
+            newConnection.slot = slot;
+            return m_connections.insert(std::move(newConnection));
+        }
+
+        // Establish a deferred connection between signal and slot, where ConnectionEvaluator object
+        // is used to queue all the connection to evaluate later. The returned
+        // value can be used to disconnect the slot later.
+        Private::GenerationalIndex connectDeferred(const std::shared_ptr<ConnectionEvaluator> &evaluator, std::function<void(const ConnectionHandle &, Args...)> const &slot)
+        {
+            auto weakEvaluator = std::weak_ptr<ConnectionEvaluator>(evaluator);
+
+            auto deferredSlot = [weakEvaluator = std::move(weakEvaluator), slot](const ConnectionHandle &handle, Args... args) {
+                if (auto evaluatorPtr = weakEvaluator.lock()) {
+                    auto lambda = [slot, handle, args...]() {
+                        slot(handle, args...); // Call slot with ConnectionHandle and Args
+                    };
+                    evaluatorPtr->enqueueSlotInvocation(handle, lambda);
+                } else {
+                    throw std::runtime_error("ConnectionEvaluator is no longer alive");
+                }
+            };
+
+            Connection newConnection;
+            newConnection.m_connectionEvaluator = evaluator;
+            newConnection.slotDeferred = deferredSlot;
+
+            return m_connections.insert(std::move(newConnection));
         }
 
         // Disconnects a previously connected function
-        void disconnect(const Private::GenerationalIndex &id) override
+        void disconnect(const Private::GenerationalIndex &id, const ConnectionHandle &handle) override
```

Comment:
Either pass in the handle and get the id out of it, or only pass the id and construct the handle on-the-fly, there's no point in having both as parameters.
The handle already contains the id...